methods.rb

```diff
@@ -0,0 +1,123 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+    i = 0
+    my_each do |item|
+      yield item, i
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+    arry = []
+    for i in self
+      arry << i if yield i
+    end
+    arry
+  end
+
+  def my_all?(arg = nil)
+    output = false
+    filtered_array = if !arg # no arguments
+                       block_given? ? my_select { |el| yield(el) } : my_select { |el| el }
+                     elsif arg.is_a?(Regexp)
+                       my_select { |el| arg.match(el) }
+                     elsif arg.is_a?(Class)
+                       my_select { |el| el.class <= arg }
+                     else
+                       my_select { |el| el == arg }
+                     end
+    output = true if filtered_array == to_a
+    output
+  end
+
+  def my_any?(arg = nil)
+    output = false
+    filtered_array = if !arg # no arguments
+                       block_given? ? my_select { |el| yield(el) } : my_select { |el| el }
+                     elsif arg.is_a?(Regexp)
+                       my_select { |el| arg.match(el) }
+                     elsif arg.is_a?(Class)
+                       my_select { |el| el.class <= arg }
+                     else
+                       my_select { |el| el == arg }
+                     end
+    output = true unless filtered_array.to_a.empty?
+    output
+  end
+
+  def my_none?(arg = nil)
+    output = false
+    filtered_array = if !arg
+                       block_given? ? my_select { |el| yield(el) } : my_select { |el| el }
+                     elsif arg.is_a?(Regexp)
+                       my_select { |el| arg.match(el) }
+                     elsif arg.is_a?(Class)
+                       my_select { |el| el.class <= arg }
+                     else
+                       my_select { |el| el == arg }
+                     end
+    output = true if filtered_array.to_a.empty?
+    output
+  end
+
+  def my_count (argument=nil)
+   counter = 0
+   if block_given?
+    to.a.my_each {|item| counter += 1 if yield(item)}
+   elsif !argument.nil?
+    to_a.my_each{|item| counter += 1 if argument == item}
+  else
+    counter = to_a.length
+  end
+  counter
+  end  
```

Comment:
#my_count
- [ ] If a block is given, it counts the number of elements yielding a true value.
![true](https://user-images.githubusercontent.com/33067256/98527585-43037d00-228c-11eb-8867-7c7d4bc6aae0.png)

- What it expects `array.count{ |x| x%2==0 } #=> 3`

- [ ] it should not change the original array
![mutate](https://user-images.githubusercontent.com/33067256/98527431-10f21b00-228c-11eb-93f7-dec07af93c75.png)